tooling/image-updater/internal/clients/acr.go

```diff
@@ -69,19 +69,29 @@ func (c *ACRClient) getAllTags(ctx context.Context, repository string) ([]Tag, e
 }
 
 func (c *ACRClient) getAllTagsWithClient(ctx context.Context, repository string, client *azcontainerregistry.Client) ([]Tag, error) {
+	logger := logr.FromContextOrDiscard(ctx)
```

Comment:
Always either explicitly pass the logger (best) or error if missing (acceptable). We never want the logs to be missing, so `OrDiscard` is never the right thing to do.